source/common/compressor/zlib_compressor_impl.cc

```diff
@@ -85,10 +85,11 @@ void ZlibCompressorImpl::process(Buffer::Instance& output_buffer, int64_t flush_
 
 void ZlibCompressorImpl::updateOutput(Buffer::Instance& output_buffer) {
   const uint64_t n_output = chunk_size_ - zstream_ptr_->avail_out;
-  if (n_output > 0) {
-    output_buffer.add(static_cast<void*>(chunk_char_ptr_.get()), n_output);
+  if (n_output == 0) {
+    return;
   }
-  chunk_char_ptr_ = std::make_unique<unsigned char[]>(chunk_size_);
+
+  output_buffer.add(static_cast<void*>(chunk_char_ptr_.get()), n_output);
   zstream_ptr_->avail_out = chunk_size_;
   zstream_ptr_->next_out = chunk_char_ptr_.get();
```

Comment:
Technically I think you can remove this line also? Maybe ASSERT it's still equal to the same pointer.

Reply:
it isn't -- zlib changes it internally... that's why I left it.

Reply:
OK thanks